### PR TITLE
bpo-39589: Context manager support for the QueueListener

### PR DESCRIPTION
**Pull request by sbrugman**
This pull request adds context manager support to the QueueListener. 

https://bugs.python.org/issue39589

### CODE REVIEW

**Review by vsajip — COMMENTED** [review]
I feel you might have misunderstood how `QueueHandler` and `QueueListener` are supposed to work, based on your code in `test_contextmanager` which uses a `with` block to do some logging.

In practice, the part that does the logging (i.e. application code) uses `QueueHandler`, not `QueueListener`. The intention is for `QueueListener` to be used in a separate thread or process whose only job is to listen to messages from the `QueueHandler` and dispatch them to appropriate destinations. In this case, as there is only one job to do, there is no special context that needs to be invoked. The logging cookbook has [an example of this usage](https://docs.python.org/3/howto/logging-cookbook.html#a-more-elaborate-multiprocessing-example).

Using handlers in context managers make sense, and there is already [a cookbook recipe](https://docs.python.org/3/howto/logging-cookbook.html#using-a-context-manager-for-selective-logging) for that. I don't want to bake in context management for any particular handler at the moment - the use cases are too varied for a definitively useful basis for design/implementation. So the cookbook serves as a guide for people to implement their own requirements using the recipe as a starting point.

In summary, I think you should close this PR and the issue as I think they are based on a misunderstanding.
